core/src/edu/cornell/gdiac/main/SpinnerModel.java

```diff
@@ -0,0 +1,102 @@
+package edu.cornell.gdiac.main;
+
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.physics.box2d.BodyDef.BodyType;
+import com.badlogic.gdx.physics.box2d.Joint;
+import com.badlogic.gdx.physics.box2d.World;
+import com.badlogic.gdx.physics.box2d.joints.RevoluteJointDef;
+import com.badlogic.gdx.utils.JsonValue;
+import edu.cornell.gdiac.assets.AssetDirectory;
+import edu.cornell.gdiac.physics.obstacle.ComplexObstacle;
+import edu.cornell.gdiac.physics.obstacle.Obstacle;
+import edu.cornell.gdiac.physics.obstacle.WheelObstacle;
+import edu.cornell.gdiac.util.TiledJsonParser;
+
+/**
+ * A BoxObstacle that represents a spinner, which can rotate around its center or a pivot point.
+ * <p>
+ * This class can be initialized from a JSON file, allowing for easy level design and modification.
+ */
+public class SpinnerModel extends ComplexObstacle {
+
+  private float defaultPivotRadius = 0.05f;
+  private float pinDensity = 10.0f;
```

Comment:
are these constants? If so indicate it by putting then in `SCREAMING_SNAKE_CASE` and make them final.